### ISSUE TITLE
Implementation in native ComfyUI workflow

### ISSUE THREAD

**Issue by nestflow** [vita-epfl/Stable-Video-Infinity#35]
Hi, appreciate for your great work and I can't wait to try the new SVI 2.0 for Wan 2.2 in ComfyUI. I read your codes and workflow on Kijai's custom nodes, and I want to get a native ComfyUI implementation. According to my understanding of native `WanImageToVideo` node,

https://github.com/comfyanonymous/ComfyUI/blob/4086acf3c2f0ca3a8861b04f6179fa9f908e3e25/comfy_extras/nodes_wan.py#L40-L60

Assume we have prepared the `start_image` to be a batch of `[last_frame, initial_image * (lengths - 1)]`, the only changes to make SVI-2.0(and shot) work should be (from line 48):

```
            mask = torch.ones((1, 1, latent.shape[2], concat_latent_image.shape[-2], concat_latent_image.shape[-1]), device=start_image.device, dtype=start_image.dtype)
            if keep_frames is None: # Original line
                mask[:, :, :((start_image.shape[0] - 1) // 4) + 1] = 0.0
            else: # Keep frames
                mask_latent_length = ((keep_frames - 1) // 4) + 1
                if mask_latent_length > mask.shape[2]:
                    mask_latent_length = mask.shape[2]
                mask[:, :, :mask_latent_length] = 0.0
```

with `keep_frames=1`. The output mask is essentially the same as the one with single frame of input image - first mask = 0 and all else = 1, and from what I read from your codes, this is exactly what we want. The original line will mask all frames = 0 since we are giving it an image batch with `length` images.

I made a testing custom node myself with the edits above, and it seems to work well. Could you help me to verify if I get it correctly? If so, I may release the custom node somewhere to help more people.

The full codes:
```
class WanImageToVideoKeepFrames(io.ComfyNode):
    @classmethod
    def define_schema(cls):
        return io.Schema(
            node_id="WanImageToVideoKeepFrames",
            category="conditioning/video_models",
            inputs=[
                io.Conditioning.Input("positive"),
                io.Conditioning.Input("negative"),
                io.Vae.Input("vae"),
                io.Int.Input("width", default=832, min=16, max=nodes.MAX_RESOLUTION, step=16),
                io.Int.Input("height", default=480, min=16, max=nodes.MAX_RESOLUTION, step=16),
                io.Int.Input("length", default=81, min=1, max=nodes.MAX_RESOLUTION, step=4),
                io.Int.Input("batch_size", default=1, min=1, max=4096),
                io.ClipVisionOutput.Input("clip_vision_output", optional=True),
                io.Image.Input("start_image", optional=True),
                io.Int.Input("keep_frames", default=1, min=1, max=nodes.MAX_RESOLUTION, step=4)
            ],
            outputs=[
                io.Conditioning.Output(display_name="positive"),
                io.Conditioning.Output(display_name="negative"),
                io.Latent.Output(display_name="latent"),
            ],
        )

    @classmethod
    def execute(cls, positive, negative, vae, width, height, length, batch_size, start_image=None, clip_vision_output=None, keep_frames=None) -> io.NodeOutput:
        latent = torch.zeros([batch_size, 16, ((length - 1) // 4) + 1, height // 8, width // 8], device=comfy.model_management.intermediate_device())
        if start_image is not None:
            start_image = comfy.utils.common_upscale(start_image[:length].movedim(-1, 1), width, height, "bilinear", "center").movedim(1, -1)
            image = torch.ones((length, height, width, start_image.shape[-1]), device=start_image.device, dtype=start_image.dtype) * 0.5
            image[:start_image.shape[0]] = start_image

            concat_latent_image = vae.encode(image[:, :, :, :3])
            mask = torch.ones((1, 1, latent.shape[2], concat_latent_image.shape[-2], concat_latent_image.shape[-1]), device=start_image.device, dtype=start_image.dtype)
            if keep_frames is None:
                mask[:, :, :((start_image.shape[0] - 1) // 4) + 1] = 0.0
            else:
                mask_latent_length = ((keep_frames - 1) // 4) + 1
                if mask_latent_length > mask.shape[2]:
                    mask_latent_length = mask.shape[2]
                mask[:, :, :mask_latent_length] = 0.0

            positive = node_helpers.conditioning_set_values(positive, {"concat_latent_image": concat_latent_image, "concat_mask": mask})
            negative = node_helpers.conditioning_set_values(negative, {"concat_latent_image": concat_latent_image, "concat_mask": mask})

        if clip_vision_output is not None:
            positive = node_helpers.conditioning_set_values(positive, {"clip_vision_output": clip_vision_output})
            negative = node_helpers.conditioning_set_values(negative, {"clip_vision_output": clip_vision_output})

        out_latent = {}
        out_latent["samples"] = latent
        return io.NodeOutput(positive, negative, out_latent)
```

**Comment by armynew** [comment]
> Thanks for your valuable contributions! We have released a preview workflow for SVI + Wan 2.2, available [here](https://github.com/vita-epfl/Stable-Video-Infinity/blob/svi_wan22/comfyui_workflow/SVI-Wan22-1207.json). This workflow is built on [ComfyUI-WanVideoWrapper](https://github.com/kijai/ComfyUI-WanVideoWrapper). Feel free to refer to it for your implementation. In addition, the native workflow is in progress.

I use your WF, the result  back to the origin ref image

https://github.com/user-attachments/assets/792dccfe-8f2b-4ba5-b21d-909b0b3494d0

this is ref image
<img width="320" height="544" alt="Image" src="https://github.com/user-attachments/assets/71d6ad5e-5716-41da-a971-d1e29df58371" />

**Comment by armynew** [comment]
Thank you for your explain

**Comment by wymanCV** [comment]
@armynew Hi, we just found another critical cause related to your issue. The LightX2V LoRA scale also has a significant impact. With the default value of 1.0, we see clearly worse dynamics and text-following compared to runs with a reduced scale (this effect is not caused by SVI itself). However, lowering the LightX2V LoRA weight will also affect the generation quality when using 4/8-step sampling. So this is a trade-off: it’s better to try adjusting this value first if you notice poor text-following, weak dynamics, or the reference frame reappearing (0.6 seems to be a good option for this sample). Thanks.

https://github.com/user-attachments/assets/fb5a33f6-fd1f-433f-9a10-ba09982ae8f1

**Comment by wymanCV** [comment]
We have updated the workflow to fix the bugs related to the LightX2V LoRA (see the [README](https://github.com/vita-epfl/Stable-Video-Infinity/tree/svi_wan22) for details). Using the new workflow, it should now be able to reproduce this video.

https://github.com/user-attachments/assets/c42aae8f-c594-4d59-82be-ed03ccd78682

**Comment by tester4488** [comment]
<img width="1382" height="877" alt="Image" src="https://github.com/user-attachments/assets/60374911-5e95-4812-ba4b-b4b696539ebc" />

the second part, only produce fixed noise image

**Comment by SRStwo** [comment]
I am not an expert at all but I have stopped using lightning/light2x accelerators with Wan 2.2 for a lot of my video work because the quality is so much lower with them.

If you can provide, at some future time, guidance/workflow for native ComfyUI that does not incorporate those accelerator loras that would be very appreciated. Thank you.
